### DISCUSSION TITLE
Getting reading errors most of the time

### DISCUSSION THREAD

**Discussion by pbuttelli** [chillerlan/php-qrcode#174] [Q&A] [answered]
I'm trying to generate a QR Code from an otpauth URI (like below). Sometimes, the code generates a valid image, that can be read by my phone. However, most of the time it outputs an invalid code.

```
otpauth://totp/Tester:tester@gmail.com?secret=A63BWG5IWX657J62V6UGFQPLVR4N636L&issuer=Tester
```

![image](https://user-images.githubusercontent.com/11038642/209169322-f1aa429e-56bc-455f-84be-cde91ab0ca20.png)

Parsed result:
```
otpauth://totp/Tester:tester@gmail.com?secret=A62�bu�?��[���!���P��D�<�����d�H{E����:�(t��
```

Is there any way to solve this?

Thanks in advance,
Pedro

**Comment by codemasher** [comment]
Hey, thanks for the report! I checked back with dev-main and cannot reproduce the error. It would be very helpful if you could post the code that produced the above QR code.

**Comment by pbuttelli** [comment, marked as answer]
Thank you for your reply, @codemasher!

I've figured out what I was doing wrong. I was encrypting data before sending to the image generator file (`qrcode.php`) via GET parameters. When decrypting, I mistakenly used the param instead of the decrypted string, which explains why I was seeing a different QR Code every page refresh. A stupid error that caused me headaches for hours (laughs).

Now, everything seems to work fine.

Merry Christmas!
Pedro
